backend/backend.ts

```diff
@@ -194,6 +211,65 @@ app.get('/load', authenticate, async (req: AuthenticatedRequest & TypedRequestBo
   }
 });
 
+// Share link create endpoint
+app.post('/share', optionalAuthenticate, shareRateLimit, async (req: AuthenticatedRequest & TypedRequestBody<{ data: any }>, res: Express.Response) => {
+  try {
+    const { username } = req.user as jwt.JwtPayload & { username: string };
+    const factoryData = req.body;
+
+    console.log(`Creating share link for user ${username}`);
+
+    const shareId = await generateShareWords(3);
+
+    const shareData: ShareDataSchema = {
+      id: shareId,
+      data: JSON.stringify(factoryData),
```

Comment:
Perhaps an enhancement, but we could use `messagePack` here to compress the JSON.  
> According to tests and comparisons, MessagePack can compress data by up to 50% compared to JSON. This means a 10,000 character JSON string could potentially be reduced to around 5,000 characters when using MessagePack.

Reply:
I like the suggestion, I'll look into implementing it as it would be nice to cut down on any storage.

Reply:
created #89 to track

Reply:
@samsmithnz Unfortunately it's using buffers, and with typescript buffers are a MASSIVE pain in the ass to get right and it be happy with it, so on this occasion I'm going to value my sanity over data storage, on a 12TB NAS pretty sure this will never be a problem :D